src/common.cpp

```diff
@@ -254,20 +256,23 @@ namespace irods
         using irods::experimental::filesystem::perms;
 
         switch (_p) {
+            // clang-format off
             case perms::null:            return "null";
             case perms::read_metadata:   return "read_metadata";
-            case perms::read_object:     return "read_object";
+            case perms::read_object:
             case perms::read:            return "read_object";
             case perms::create_metadata: return "create_metadata";
             case perms::modify_metadata: return "modify_metadata";
             case perms::delete_metadata: return "delete_metadata";
             case perms::create_object:   return "create_object";
-            case perms::modify_object:   return "modify_object";
+            case perms::modify_object:
             case perms::write:           return "modify_object";
             case perms::delete_object:   return "delete_object";
             case perms::own:             return "own";
-            default:                     return "?"; // TODO std::unreachable() or __builtin_unreachable()
+            // clang-format on
         }
+
+        THROW(SYS_INVALID_INPUT_PARAM, fmt::format("Cannot convert permission enumeration to string."));
```

Comment:
any way to write/show what it tried to do?

Reply:
Yes. Will adjust.

Reply:
Actually, there's no way to do that since we're given an enumeration.

We should never reach the `THROW` unless we do something suspicious like cast an integer to an enumeration.